funcx_sdk/funcx/serialize/concretes.py

```diff
@@ -124,16 +128,87 @@ class DillCode(SerializeBase):
     def __init__(self):
         super().__init__()
 
-    def serialize(self, data):
+    def serialize(self, data) -> str:
         x = codecs.encode(dill.dumps(data), "base64").decode()
         return self.identifier + x
 
-    def deserialize(self, payload):
+    def deserialize(self, payload: str):
         chomped = self.chomp(payload)
         function = dill.loads(codecs.decode(chomped.encode(), "base64"))
         return function
 
 
+class CombinedCode(SerializeBase):
+    """This method uses multiple methods to serialize a function
+
+    # These should be a superset of the methods
+    Code from interpreter/main        : Yes
+    Code from notebooks               : Yes
+    Works with mismatching py versions: Yes
+    Decorated fns                     : Yes
+    """
+
+    identifier = "10\n"
+
+    def get_multiple_payloads(self, payload: str) -> Tuple:
+        parts = self.chomp(payload).split(COMBINED_SEPARATOR)
+        assert len(parts) == 2 * len(DO_SERIALIZE_METHODS)
+        payloads = []
+        for index in range(len(parts) // 2):
+            payloads.append([parts[index], parts[index + 1]])
+
+        # ie. [['04\n', 'gACQ...'], ['01\n', 'sAAbk2...'], ...]
+        return payloads
```

Comment:
It Does. Not. Matter. for such a small list, but this makes me think of an alternative approach.  Consider:

```python
parts_i = iter(self.chomp(payload).split(COMBINED_SEPARATOR))
return tuple(zip(parts_i, parts_i))
```

That ignores whether the list is an even-number of elements or not -- let the caller handle the error if it's not (the last element of an odd-number of items will be ignored).

Reply:
I'm leaning toward having it checked/handled here, as storing/reading it in this [[id_a, payload_a], [id_b, payload_b]...] format is the implementation details of CombinedCode and should be opaque to callers